src/main/java/com/iota/iri/IRI.java

```diff
@@ -178,6 +179,23 @@ private static IotaConfig createConfiguration(String[] args) {
             return iotaConfig;
         }
 
+        /**
+         * We are connected to testnet when {@link Config#TESTNET_FLAG} is passed in program startup,
+         * following with <code>true</code>
+         * 
+         * @param args the list of program startup arguments
+         * @return <code>true</code> if this is testnet, otherwise <code>false</code>
+         */
+        private static boolean isTestnet(String[] args) {
```

Comment:
What about parsing the boolean value passed to options that are not `--testnet`. Is `--remote false` treated correctly?

Reply:
No all other parsing happens in JCommander, but this is needed because we instantiate a different config based on testnet, so we have to check that Before using jcommander.. silly

Removing the testnet/mainnet difference is a different issue/pr i think